### PR TITLE
Fix shellcheck failures in stage-upload.sh

### PR DESCRIPTION
**Pull request by xichengliudui**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Ref ##72956

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cluster/gce/gci/mounter/stage-upload.sh [hunk 1]
```diff
@@ -39,29 +39,29 @@ ACI_DIR=${STAGING_DIR}/gci-mounter
...
 # Delete temporary directories on exit
 trap cleanup EXIT
 
-mkdir ${ACI_DIR}
+mkdir "${ACI_DIR}"
 
 # Convert docker image to aci and stage it
 echo "Downloading docker2aci ${DOCKER2ACI_VERSION}"
-wget "https://github.com/appc/docker2aci/releases/download/${DOCKER2ACI_VERSION}/docker2aci-${DOCKER2ACI_VERSION}.tar.gz" &> /dev/null
+wget "https://github.com/appc/docker2aci/releases/download/${DOCKER2ACI_VERSION}/docker2aci-${DOCKER2ACI_VERSION}.tar.gz" "&>" /dev/null
```
Comment (xichengliudui) [inline cluster/gce/gci/mounter/stage-upload.sh]: /assign @BenTheElder 
Is this correct? I'm not sure.

Comment (xichengliudui) [reply cluster/gce/gci/mounter/stage-upload.sh]:
```
In ./cluster/gce/gci/mounter/stage-upload.sh line 55:
wget "https://github.com/appc/docker2aci/releases/download/${DOCKER2ACI_VERSION}/docker2aci-${DOCKER2ACI_VERSION}.tar.gz" &> /dev/null
                                                          ^----------^ SC2039: In POSIX sh, &> is undefined.
Ref: https://github.com/koalaman/shellcheck/wiki/SC2039
```

Comment (BenTheElder) [reply cluster/gce/gci/mounter/stage-upload.sh]: This is not correct, `&> /dev/null` is shell redirection in bash to redirect both stdout and stderr to `/dev/null`. In POSIX sh `>/dev/null 2>&1` is the correct equivilant.

Comment (xichengliudui) [reply cluster/gce/gci/mounter/stage-upload.sh]: YES, but why 2>&1?

Comment (xichengliudui) [reply cluster/gce/gci/mounter/stage-upload.sh]: has been updated.

Comment (BenTheElder) [reply cluster/gce/gci/mounter/stage-upload.sh]: the file descriptors for stdin, stdout, and stderr have well known numbers: https://en.wikipedia.org/wiki/File_descriptor

`2>&1` redirects stderr to stdout, `>/dev/null 2>&1` redirects stdout to `/dev/null` and then stderr to stdout.

Comment (xichengliudui) [reply cluster/gce/gci/mounter/stage-upload.sh]: has been updated.

File: cluster/gce/gci/mounter/stage-upload.sh [hunk 2]
```diff
@@ -39,29 +39,29 @@ ACI_DIR=${STAGING_DIR}/gci-mounter
...
 tar xzf docker2aci-${DOCKER2ACI_VERSION}.tar.gz
-ACI_IMAGE=$(${DOWNLOAD_DIR}/docker2aci-${DOCKER2ACI_VERSION}/docker2aci ${DOCKER_IMAGE} 2>/dev/null | tail -n 1)
-cp ${ACI_IMAGE} ${ACI_DIR}/${MOUNTER_ACI_IMAGE}
+ACI_IMAGE=$("${DOWNLOAD_DIR}"/docker2aci-"${DOCKER2ACI_VERSION}"/docker2aci "${DOCKER_IMAGE}" 2>/dev/null | tail -n 1)
+cp "${ACI_IMAGE}" "${ACI_DIR}/${MOUNTER_ACI_IMAGE}"
 
 # Upload the contents to gcs
 echo "Uploading gci mounter ACI in ${ACI_DIR} to ${MOUNTER_GCS_DIR}"
-gsutil cp ${ACI_DIR}/${MOUNTER_ACI_IMAGE} ${MOUNTER_GCS_DIR}
+gsutil cp "${ACI_DIR}"/"${MOUNTER_ACI_IMAGE}" "${MOUNTER_GCS_DIR}"
```
Comment (filbranden) [inline cluster/gce/gci/mounter/stage-upload.sh]: Using a single string for the src `"${ACI_DIR}/${MOUNTER_ACI_IMAGE}"` would have been fine as well...

File: cluster/gce/gci/mounter/stage-upload.sh [hunk 3]
```diff
@@ -39,29 +39,29 @@ ACI_DIR=${STAGING_DIR}/gci-mounter
...
 
 echo "Upload completed"
 echo "Updated gci-mounter ACI version and SHA1 in cluster/gce/gci/configure.sh"
-echo "${MOUNTER_ACI_IMAGE} hash: $(sha1sum ${ACI_DIR}/${MOUNTER_ACI_IMAGE})"
+echo "${MOUNTER_ACI_IMAGE}" hash: "$(sha1sum "${ACI_DIR}"/"${MOUNTER_ACI_IMAGE}")"
```
Comment (filbranden) [inline cluster/gce/gci/mounter/stage-upload.sh]: Here I'd recommend using a separate variable to store the hash:

```sh
ACI_HASH=$(sha1sum "${ACI_DIR}/${MOUNTER_ACI_IMAGE}")
echo "${MOUNTER_ACI_IMAGE} hash: ${ACI_HASH}"
```

**Review by filbranden — APPROVED** [review]
Looks good overall. I think the second comment (about the hash) deserves to be addressed, since having `echo` take a single string is better I'd say (also closer to original command.)

File: cluster/gce/gci/mounter/stage-upload.sh (hunk 3 above)
Comment (xichengliudui) [reply cluster/gce/gci/mounter/stage-upload.sh]: YES, has been updated.

**Review by filbranden — APPROVED** [review]
/lgtm

File: cluster/gce/gci/mounter/stage-upload.sh [hunk 4]
```diff
@@ -39,29 +39,30 @@ ACI_DIR=${STAGING_DIR}/gci-mounter
...
 
 # Convert docker image to aci and stage it
 echo "Downloading docker2aci ${DOCKER2ACI_VERSION}"
-wget "https://github.com/appc/docker2aci/releases/download/${DOCKER2ACI_VERSION}/docker2aci-${DOCKER2ACI_VERSION}.tar.gz" &> /dev/null
+wget "https://github.com/appc/docker2aci/releases/download/${DOCKER2ACI_VERSION}/docker2aci-${DOCKER2ACI_VERSION}.tar.gz" >/dev/null 2>&1
 echo "Extracting docker2aci ${DOCKER2ACI_VERSION}"
 tar xzf docker2aci-${DOCKER2ACI_VERSION}.tar.gz
-ACI_IMAGE=$(${DOWNLOAD_DIR}/docker2aci-${DOCKER2ACI_VERSION}/docker2aci ${DOCKER_IMAGE} 2>/dev/null | tail -n 1)
-cp ${ACI_IMAGE} ${ACI_DIR}/${MOUNTER_ACI_IMAGE}
+ACI_IMAGE=$("${DOWNLOAD_DIR}"/docker2aci-"${DOCKER2ACI_VERSION}"/docker2aci "${DOCKER_IMAGE}" 2>/dev/null | tail -n 1)
```
Comment (cblecker) [inline cluster/gce/gci/mounter/stage-upload.sh]: quote the whole filename, not the individual vars
```suggestion
ACI_IMAGE=$("${DOWNLOAD_DIR}/docker2aci-${DOCKER2ACI_VERSION}/docker2aci" "${DOCKER_IMAGE}" 2>/dev/null | tail -n 1)
```

**Review by cblecker — CHANGES_REQUESTED** [review]

File: cluster/gce/gci/mounter/stage-upload.sh (hunk 4 above)
Comment (xichengliudui) [reply cluster/gce/gci/mounter/stage-upload.sh]: has been updated.

**Review by cblecker — APPROVED** [review]
/lgtm
/approve

**Comment by cblecker** [comment]
@filbranden if things look good to you, can you `/approve` this?

**Comment by filbranden** [comment]
/approve

It sure does. Thanks @cblecker !

**Pull request merged**